packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -283,6 +287,28 @@ export class SharedTree
 		);
 	}
 
+	public branch(): TreeBranch {
```

Comment:
Do we want anyone doing operations like this directly on the SharedTree class, rather than just on checkouts?

Reply:
I think so. Currently we don't expose the concept of a "checkout". My thinking was that if you want to make a branch, you just do `sharedTree.branch()` rather than `sharedTree.checkout.branch()` or something. Just to keep it simple. The thing that you get from the call to `branch()` (the branch) is similar to a SharedTree - for example, it has `viewWith` (and later maybe we add `getContentSnapshot` and whatever else). Basically, it has all the stuff that a SharedTree has, except for the Fluid SharedObject stuff. Introducing a checkout concept isn't horrible but I feel it might just add another concept without any benefit.

However, we can't actually write the API like that and have it be _alpha_, because SharedTree is public. So, for now, you still have to pass the SharedTree to `getBranch()` (which is sort of like "getting a checkout"). But I want the eventual public API to be on the SharedTree itself, so I have the methods here in anticipation of that (they are not exposed on the ITree interface, for the reasons stated above).

Reply:
Discussed offline - we pivoted to a strategy where TreeCheckouts are directly provided as the branches (via a reduced interface).